### PR TITLE
Feature/auto update

### PR DESCRIPTION
**Pull request by agduncan94**
This adds the ability to upgrade the CLI to the latest stable version (tag on Github). Also more information  is given from the --version flag.

Issue #66

### CODE REVIEW

**Comment by agduncan94** [comment]
A strange thing I noticed was that cwltool was throwing errors if I had a direct path to dockstore in the PATH variable (.../dockstore-client/target/dockstore).  Changing to .../dockstore-client/target stopped the error from occurring.

**Comment by denis-yuen** [comment]
I think we're going to have to sit down and explore git squash together, especially for commits that are mostly tinkering with .travis.yml

File: dockstore-client/src/main/java/io/dockstore/client/cli/Client.java [hunk 1]
```diff
@@ -1177,6 +1194,256 @@ public static void updateContainerHelp() {
...
+
+        return installLocation;
+    }
+
+    /**
+     * Finds the version of the dockstore CLI for the given install location
+     * @param installLocation
+     * @return
+         */
+    public static String getCurrentVersion(String installLocation){
```
Comment (denis-yuen) [inline dockstore-client/src/main/java/io/dockstore/client/cli/Client.java]: Please clarify that this is to get the script version (as opposed to the version in the jar which can be obtained more simply). On the other hand, a thought about simplification occurs. Is there any situation in which the script and the jar have different versions? If there is none, then consider just getting the version from the jar which is simpler ex: https://github.com/SeqWare/seqware/blob/50d2ed3c449e88e43232e1af8a957ec8414441bd/seqware-webservice/src/main/java/net/sourceforge/seqware/webservice/resources/queries/EnvironmentResource.java#L35

File: dockstore-client/src/main/java/io/dockstore/client/cli/Client.java [hunk 2]
```diff
@@ -1177,6 +1194,256 @@ public static void updateContainerHelp() {
...
+        URL url = null;
+        String latestPath = "https://api.github.com/repos/ga4gh/dockstore/releases/latest";
+        String latestVersion = null;
+        try {
+            url = new URL(latestPath);
+            ObjectMapper mapper = new ObjectMapper();
+            Map<String, Object> map = null;
+            try {
+                // Read JSON from Github
+                map = mapper.readValue(url, Map.class);
```
Comment (denis-yuen) [inline dockstore-client/src/main/java/io/dockstore/client/cli/Client.java]: There is a library for talking to git, give it a shot. 

ex: https://github.com/ga4gh/dockstore/blob/783f1ccb95a37e62865d3a00170a65738b639c72/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

**Pull request merged**